xonsh: 0.8.3 -> 0.8.12

Meptl opened the pull request:
* Expanded unit tests.
 * Added pytest-rerunfailures due to flaky tests.

###### Motivation for this change
Failing hydra build.
https://hydra.nixos.org/build/91523333
/cc https://github.com/NixOS/nixpkgs/issues/56826

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

###### Tests
~~Note that to build this with a local master you need a small edit to pytest-rerunfailures. Add pytest to checkInputs. See part of https://github.com/NixOS/nixpkgs/pull/58710.~~ <- PR released
 * nix-build -A xonsh
 * nix-shell -I nixpkgs=/path/to/nixpkgs xonsh --pure -p xonsh

Meptl committed:
xonsh: 0.8.3 -> 0.8.12

Meptl commented on pkgs/shells/xonsh/default.nix:
```diff
@@ -2,31 +2,34 @@
   postPatch = ''
     sed -ie "s|/bin/ls|${coreutils}/bin/ls|" tests/test_execer.py
-    sed -ie 's|/usr/bin/env|${coreutils}/bin/env|' scripts/xon.sh
+    sed -ie "s|SHELL=xonsh|SHELL=$out/bin/xonsh|" tests/test_integrations.py
```
I'm using $out/bin/xonsh because (if I understand correctly) for python projects the checkPhase == installCheckPhase, so we want to reference the "installed" binary.

Meptl commented on pkgs/shells/xonsh/default.nix:
```diff
@@ -2,31 +2,34 @@
-        -k 'not test_man_completion and not test_indir and not test_xonsh_party and not test_foreign_bash_data and not test_script and not test_single_command_no_windows and not test_redirect_out_to_file and not test_sourcefile and not test_printname and not test_printfile'
+    HOME=$TMPDIR pytest -k 'not test_repath_backslash and not test_os and not test_man_completion and not test_builtins and not test_main and not test_ptk_highlight'
+    HOME=$TMPDIR pytest -k 'test_builtins or test_main' --reruns 5
+    HOME=$TMPDIR pytest -k 'test_ptk_highlight'
```
Running this one separately is something that xonsh does as well. See xonsh/run-tests.xsh.

Ma27 commented:
@GrahamcOfBorg build xonsh

Ma27 reviewed (approved):
<del>The build broke locally due to `python.pkgs.pytest-rerunfailures` being broken, I just pushed a fix. As soon as ofBorg finished successfully, I'd merge and backport :+1: </del>

Just seen that it's on master already, but the PR contains an old state.

Ma27 commented:
@GrahamcOfBorg build xonsh

Ma27 commented:
Just a short update: I removed my commit again as I missed that `pytest-rerunfailures` is already fixed on master. I'm a bit hesitant now to merge this and probably leave the darwin build broken. I'll see if I can bootstrap a testing environment for this locally and report back.

Meptl commented:
I don't have a mac myself, but I was able to run some tests through travis. https://travis-ci.org/Meptl/nixpkgs/builds/516748079 (darwin-test branch). Copied the nix-build command from OfBorg README. I don't think the most recent travis osx image is nix-ready. Still investigating the travis environment to see how I might reproduce.

Ma27 commented:
> I don't have a mac myself, but I was able to run some tests through travis

Thanks a lot for doing this! I don't own a mac either, I planned to test this in a virtualbox vm with osx today, but you were faster regarding a macos test :sweat_smile: 

The only thing I'm not sure about atm is whether or not to keep the tests in darwin or to disable those with `doCheck = !stdenv.isDarwin`.

@grahamc would you mind having a look at the ofborg errors for darwin? Do you think that this is related to ofborg's testing environment or should we entirely disable the tests on darwin for now? :)

timokau commented:
Since xonsh is currently broken on master, should we just exclude darwin for now? If there is interest from a mac user, they are in a much better position to fix it.

Ma27 commented:
That makes sense, yes :+1: 

This should be backported as well then, IIRC the package is broken on `release-19.03` as well.

Meptl commented:
Rebased off latest master, added the doCheck. I force pushed twice because I used the wrong email as the commit author. Kicked off OSX tests for every image available on travis: https://travis-ci.org/Meptl/nixpkgs/builds/519768510 (darwin-test2 branch)

timokau commented:
@GrahamcOfBorg build xonsh

Ma27 commented:
@timokau can you backport to `release-19.03`, please? :)

timokau commented:
https://github.com/NixOS/nixpkgs/pull/59509
